### PR TITLE
Clean build from winget-cli

### PR DESCRIPTION
**Pull request by msftrubengu**
This PR allows winget-cli to be built cleanly from a new machine by importing the .vsconfig into their Visual Studio Edition

- Update .vsconfig to remove/add components.
- Add missing dependency of AppInstallerTestExeInstaller in AppInstallerCLITests
- Move AppInstallerCLITests to the top of the solution file to automatically be the first one selected on Visual Studio.
- Add nuget.config for public dependencies.

To fully build winget-cli [Microsoft Visual Studio Installer](https://marketplace.visualstudio.com/items?itemName=VisualStudioClient.MicrosoftVisualStudio2022InstallerProjects) vsix is still required.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-cli/pull/2698)

### CODE REVIEW

File: .vsconfig [hunk 1]
```diff
@@ -3,54 +3,50 @@
...
-    "Microsoft.VisualStudio.Component.AppInsights.Tools",
     "Microsoft.VisualStudio.Component.Debugger.JustInTime",
-    "Component.Microsoft.VisualStudio.LiveShare",
+    "Component.Microsoft.VisualStudio.LiveShare.2022",
     "Microsoft.VisualStudio.Component.IntelliCode",
     "Microsoft.VisualStudio.Component.VC.CoreIde",
     "Microsoft.VisualStudio.Component.VC.Tools.x86.x64",
     "Microsoft.VisualStudio.Component.Graphics.Tools",
     "Microsoft.VisualStudio.Component.VC.DiagnosticTools",
+    "Microsoft.VisualStudio.Component.Windows10SDK.19041",
```
Comment (yao-msft) [inline .vsconfig:37]: Should this be 22000 if all our projects are targeting 22000

File: src/AppInstallerCLI.sln [hunk 2]
```diff
@@ -2,10 +2,17 @@ Microsoft Visual Studio Solution File, Format Version 12.00
...
 VisualStudioVersion = 17.2.32630.192
 MinimumVisualStudioVersion = 10.0.40219.1
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "AppInstallerCLITests", "AppInstallerCLITests\AppInstallerCLITests.vcxproj", "{89B1AAB4-2BBC-4B65-9ED7-A01D5CF88230}"
+	ProjectSection(ProjectDependencies) = postProject
+		{6AA3791A-0713-4548-A357-87A323E7AC3A} = {6AA3791A-0713-4548-A357-87A323E7AC3A}
+		{6CB84692-5994-407D-B9BD-9216AF77FE83} = {6CB84692-5994-407D-B9BD-9216AF77FE83}
+		{82B39FDA-E86B-4713-A873-9D56DE00247A} = {82B39FDA-E86B-4713-A873-9D56DE00247A}
+	EndProjectSection
+EndProject
 Project("{C7167F0D-BC9F-4E6E-AFE1-012C56B48DB5}") = "AppInstallerCLIPackage", "AppInstallerCLIPackage\AppInstallerCLIPackage.wapproj", "{6AA3791A-0713-4548-A357-87A323E7AC3A}"
```
Comment (yao-msft) [inline src/AppInstallerCLI.sln:12]: nit: I think making AppInstallerCLIPackage as default still makes sense as most of the times we'll test the test package, or make AppInstallerExe as default is better than the unit tests as default. Just my experience.

**Review by yao-msft — APPROVED** [review]

File: src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/UndockedRegFreeWinRT.vcxproj [hunk 3]
```diff
@@ -23,32 +23,32 @@
...
     <RootNamespace>UndockedRegFreeWinRT</RootNamespace>
-    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersion>10.0.22000.0</WindowsTargetPlatformVersion>
   </PropertyGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'" Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v143</PlatformToolset>
```
Comment (yao-msft) [inline src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/UndockedRegFreeWinRT.vcxproj:32]: I just encountered this build failure after updating VS2022 too. I noticed for other projects, we use below to support multiple toolsets, we should do this in this project and below project too.

```
    <PlatformToolset>v140</PlatformToolset>
    <PlatformToolset Condition="'$(VisualStudioVersion)' == '15.0'">v141</PlatformToolset>
    <PlatformToolset Condition="'$(VisualStudioVersion)' == '16.0'">v142</PlatformToolset>
    <PlatformToolset Condition="'$(VisualStudioVersion)' == '17.0'">v143</PlatformToolset>
```

**Comment by yao-msft** [comment]
Binskim task behavior changed, we'll need to pull in this pr after it's merged for the build fix https://github.com/microsoft/winget-cli/pull/2704

File: src/AppInstallerCLI.sln (hunk 2 above)
Comment (marco273637288) [reply src/AppInstallerCLI.sln:5]: good

**Pull request merged**